deepspeed/comm/ccl.py

```diff
@@ -170,7 +186,7 @@ def get_all_ranks_from_group(self, group):
             while True:
                 results.append(super(CCLBackend, self).get_global_rank(group, rank))
                 rank += 1
-        except ValueError:
+        except (ValueError, RuntimeError):
```

Comment:
What is the runtime error that we can hit here?

Reply:
The `while True:` loop would iterate rank and collect global rank until rank is out of local rank range.  In older version of PyTorch, this out-of-range will throw a ValueError.  In PyTorch 2, this behavior will throw a RuntimeError.

@Liangliang-Ma